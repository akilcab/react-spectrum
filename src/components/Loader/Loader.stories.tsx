import { Story, Meta } from "@storybook/react/types-6-0";
import { Loader, LoaderProps } from "./Loader";
// @ts-ignore
import SvgEcollect from "./assets/logo.svg";

export default {
  title: "Loader",
  component: Loader,
  description: `A Loader.`,
  argTypes: {},
} as Meta;

//👇 We create a “template” of how args map to rendering
const Template: Story<LoaderProps> = (args) => <Loader {...args} />;

//👇 Each story then reuses that template
export const Default = Template.bind({});
Default.args = {
  children: <img src={SvgEcollect} alt="" />,
};
