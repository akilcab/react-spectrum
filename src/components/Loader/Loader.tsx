import React from "react";
import { Box, CircularProgress } from "@material-ui/core";

export interface LoaderProps {
  children?: React.ReactNode;
  size?: string | number;
}

export const Loader: React.FC = ({ children, size = 100 }: LoaderProps) => {
  return (
    <Box
      height="100vh"
      display="flex"
      justifyContent="center"
      alignItems="center"
    >
      <Box position="relative" display="inline-flex">
        <CircularProgress color="secondary" size={size} thickness={1} />
        <Box
          top={0}
          left={0}
          bottom={0}
          right={0}
          position="absolute"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          {children}
        </Box>
      </Box>
    </Box>
  );
};
