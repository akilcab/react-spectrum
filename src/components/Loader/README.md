# Description of component

Loaders inform the user of loading or computing processes being carried out by the system. The progress can either be
displayed with specific details or communicated as undefined by means of a repeated animation.

The size of the loader components depends on the basic type size used, which can adapt to the viewport width in
question. As such, the size of the components is flexible.

#### Example :

## How to import

```node
import Loader from "app/shared/components/Loader";
```

## How to use

```node
....
  <React.Suspense fallback={<Loader children={<img src={Logo} />} />}>
....
```

## Props

| Props       | Required | Description | Type |
| :---        | :---     | :---        |:---  |
| `children` | false | The content of the button. | `node` |
| `size` | false | Allows you to define the foreground color of a text element and its possible decorations. | `string` |

## CSS properties we can override

You can override all CSS properties like a `div` element.