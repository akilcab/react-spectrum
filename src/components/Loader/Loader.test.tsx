import renderer from "react-test-renderer";
import { Loader } from "./Loader";

describe("<Loader />", () => {
  test("rend correctement", () => {
    const component = renderer.create(<Loader />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
