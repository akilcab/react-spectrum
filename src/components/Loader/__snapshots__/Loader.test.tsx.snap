// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> rend correctement 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={21.5}
          strokeWidth={1}
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="MuiBox-root MuiBox-root-3"
    />
  </div>
</div>
`;
